NetTopologySuite/IO/ConfigurableBinaryReader.cs

```diff
@@ -0,0 +1,220 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using GeoAPI.IO;
+using NetTopologySuite.Utilities;
+
+namespace NetTopologySuite.IO
+{
+    /// <summary>
+    /// Extends the <see cref="BinaryReader" /> class to allow reading values in the specified format.    
+    /// </summary>
+    /// <remarks>
+    /// While <see cref="ConfigurableBinaryReader" /> extends <see cref="BinaryReader" /> 
+    /// adding methods for reading integer values (<see cref="ConfigurableBinaryReader.ReadInt32" />)
+    /// and double values (<see cref="ConfigurableBinaryReader.ReadDouble" />) in the specified format, 
+    /// this implementation overrides methods, such <see cref="BinaryReader.ReadInt32" /> 
+    /// and <see cref="BinaryReader.ReadDouble" /> and more, 
+    /// for reading values in the specified format.
+    /// </remarks>
+    public class ConfigurableBinaryReader : BinaryReader
```

Comment:
Can we name this something that more specifically identifies byte order as the thing you're "configuring"?

Reply:
Changed the comment. Is it okay now? Or did you mean the class name?

Reply:
Yes the class name is what I was referring to.

Reply:
Since it's a binary reader the only thing which can be changed is byte order. I don't take the `encoding` into account because we don't need it in that case. But yep, probably the name should be changed... And currently I have no idea what would be better. @roji?

Reply:
Hmmm... The ability to handle both endian methods could be expressed by BiEndianBinaryReader. But the main point of this class seems to add support for reading int/double, so maybe something like NumericBinaryReader?

Reply:
`EndianessAware[Binary]Reader` dreadfully long but describes it.
I'd say this class can be `internal`

Reply:
I would like to prefer `BiEndianBinaryReader` because it's shorter and has the same meaning as `EndianessAwareBinaryReader`. Fine?

It can't be internal since `PostGisReader` uses it.